### DISCUSSION TITLE
Persistence with rxdb

### DISCUSSION THREAD

**Discussion by kurtg-tmQ** [maxnowack/signaldb#116] [Q&A] [answered]
Do you have any examples on how to implement multiple collections?

**Comment by maxnowack** [comment, marked as answer]
Implement multiple collections with rxdb should be really straight forward. You just have to make sure that each signaldb collection uses an own persistence adapter with a dedicated rxdb collection.
One solution would be to create a factory function to create persistence adapters for different rxdb collections.
````ts
async function getCollectionByName(name: string) {
  // return the cached rxdb collection object for the given name or create it if it does not exist
}

function createPersistenceAdapterForRxdbCollection(collectionName: string) {
  const persistence: PersistenceAdapter<{ id: string, text: string, completed: boolean }, string> = {
    register: async (onChange) => {
      const collection = await getCollectionByName(collectionName)
      collection.postInsert(() => onChange(), false)
      collection.postRemove(() => onChange(), false)
      collection.postSave(() => onChange(), false)
    },
    save: async (_items, changes) => {
      const collection = await getCollectionByName(collectionName)
      await Promise.all([
        ...changes.added.map(item => collection.insert(item)),
        ...changes.modified.map(async (item) => {
          const doc = await collection.findOne({ selector: { id: item.id } }).exec()
          if (doc) await doc.update({ $set: item })
        }),
        ...changes.removed.map(async (item) => {
          const doc = await collection.findOne({ selector: { id: item.id } }).exec()
          if (doc) await doc.remove()
        }),
      ])
    },
    load: async () => {
      const collection = await getCollectionByName(collectionName)
      const items = await collection.find().exec()
      return { items: items.map(item => item.toMutableJSON()) }
    },
  }
  return persistence
}

````
*based on the [example code](https://github.com/maxnowack/signaldb/blob/main/example/src/system/setupCollection/persistence.ts#L53-L79)*

**Reply by kurtg-tmQ** [reply]
Ok thanks. I think I tried using the same process but during the part where it tries to load the items the collections get mixed up. Anyway, thanks for the example, I'll try to explore more and apply your suggestions.

**Reply by maxnowack** [reply]
Make sure you distinguish between the collections at this point. What you are describing sounds like you are using the same collection.
Please let me know if anything is unclear or if you need any further help.

**Comment by szyrelleg-tmQ** [comment]
I encountered an error when attempting to create a collection following the provided example in this repo. 

![image](https://github.com/maxnowack/signaldb/assets/122922933/f0aac294-cf4b-4eff-a0fd-e6c711d9d4a3)

by the way i'm using meteor js

**Reply by maxnowack** [reply]
Can you share the code you're using to initialize the collection? Does the error occur directly after initialization of the collection or later when you insert/update data?
